sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/ZetaSQLQueryPlanner.java

```diff
@@ -174,28 +174,39 @@ static boolean hasOnlyJavaUdfInProjects(RelOptRuleCall x) {
           } else if (rexNode instanceof RexLiteral) {
             SqlTypeName typeName = ((RexLiteral) rexNode).getTypeName();
             switch (typeName) {
-              case NULL:
+              case DOUBLE:
               case BOOLEAN:
-              case CHAR:
-              case BINARY:
+              case VARCHAR:
+              case VARBINARY:
               case DECIMAL:
+              case TIME:
+              case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+              case TIMESTAMP:
+              case ARRAY:
```

Comment:
I don't think there are ARRAY or ROW literals. I'm pretty sure both become calls to constructors.

Reply:
done